File: 2023/java/src/test/java/org/frieder/aoc/day10/TestDay10B.java
package org.frieder.aoc.day10;

import org.frieder.aoc.day10.b.Solution10B;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay10B {
    @Test
    public void testExample() throws IOException {
        assertEquals(10, Solution10B.getResult("../resources/10/test_b.txt"), 0);
    }


    @Test
    public void testResult() throws IOException {
        assertEquals(337, Solution10B.getResult("../resources/10/input.txt"), 0);
    }
}

File: 2023/php/test/TestDay10B.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day10\b\Solution10B;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay10B extends TestCase
{

    public function testExample(): void
    {
        $this->assertSame(10, Solution10B::getResult("../resources/10/test_b.txt", TestLogger::createTestLogger(), TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(337, Solution10B::getResult("../resources/10/input.txt", TestLogger::createTestLogger(), TestLogger::createTestLogger()));
    }

}